packages/other/src/DragList.ts

```diff
@@ -0,0 +1,201 @@
+                        .style("width", rect.width + "px")
+                        .style("height", rect.height + "px")
+                        .style("font-size", "20px")
+                        .style("line-height", context.itemMinHeight() + "px")
+                        .style("padding-left", rect.height + "px")
+                        ;
+                    context._dragMemory.yOffset = rect.top - context._containerTop - context._dragMemory.startY;
+                    ghost.append("div")
+                        .attr("class", "drag-list-item-label")
+                        .text(context._dragMemory.label)
+                        ;
+                    context._dragMemory.ghost = ghost.node();
+                }
+                const ghostTop = y + context._dragMemory.yOffset;
+                context._dragMemory.ghost.style.top = ghostTop + "px";
+                let idx = 0;
+                const halfItemHeight = context.itemMinHeight() / 2;
+                const dragY = y - parentScrollTop;
+                context._dragMemory.siblingTops.forEach((top, i) => {
+                    if (i === 0) {
+                        if (dragY < top - halfItemHeight) {
+                            idx = i;
+                        }
+                    }
+                    if (dragY > top - halfItemHeight) {
+                        idx = i + 1;
+                    }
+                });
+                if (context._dragMemory.idx !== idx) {
+                    context._dragMemory.idx = idx;
+                    context._container.node().children[idx].before(context._dragMemory.element);
+                }
+            }
+        });
+        element.on("mouseup", function() {
+            if (context._dragMemory.ghost) {
+                context._dragMemory.element.style.visibility = "visible";
+                context._dragMemory.element = undefined;
+                context._container.node().removeChild(context._dragMemory.ghost);
+                context._dragMemory.ghost = undefined;
+                context.applySortOrder();
+            }
+        });
+    }
+    update(domNode, element) {
+        super.update(domNode, element);
+
+        const context = this;
+
+        this._container
+            .style("min-height", (this.widgets().length + 1) * (this.itemMinHeight() + this.spaceBetweenItems()) + "px")
+            ;
+
+        const rowSelection = this._container.selectAll(".drag-list-item").data(this.widgets(), d => d.id());
+        const rowEnter = rowSelection.enter()
+            .append("div")
+            .attr("class", "drag-list-item")
+            .style("min-height", this.itemMinHeight() + "px")
+            .style("display", "flex")
+            .style("background-color", "#ddd")
+            .style("margin-bottom", this.spaceBetweenItems() + "px")
+            .style("-ms-user-select", "none")
+            .style("user-select", "none")
+            .each(function(d: any) {
```

Comment:
`d` is not an any, its a Widget and you can omit the type as TS will automatically "know"